Immich - postgresql-16-vchord - Misuse of shell builtins

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

Immich

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/immich.sh)"

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

pve-manager/9.1.1/42db4a6cf33dac83 (running kernel: 6.17.2-1-pve)

### 📝 Provide a clear and concise description of the issue.

Immich fails to install

### 🔄 Steps to reproduce the issue.

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/immich.sh)"

### ❌ Paste the full error output (if available).

================================================================================
PHASE 2: APPLICATION INSTALLATION (Container)
================================================================================
[2026-02-28 12:05:52] [INFO] Setting up Container OS
[2026-02-28 12:05:52] [OK] Set up Container OS
[2026-02-28 12:05:52] [OK] Network Connected: 192.168.0.21 fd8a:6de9:12ce:470a:be24:11ff:fef0:e6d1
[2026-02-28 12:05:53] [OK] IPv4 Internet Connected
[2026-02-28 12:05:53] [ERROR] IPv6 Internet Not Connected
[2026-02-28 12:05:53] [OK] Git DNS: github.com:(✔️ ) raw.githubusercontent.com:(✔️ ) api.github.com:(✔️ ) git.community-scripts.org:(✔️ )
[2026-02-28 12:05:53] [INFO] Updating Container OS
Hit:1 http://security.debian.org trixie-security InRelease
Hit:2 http://deb.debian.org/debian trixie InRelease
Hit:3 http://deb.debian.org/debian trixie-updates InRelease
Reading package lists...
[2026-02-28 12:06:44] [OK] Updated Container OS
[2026-02-28 12:06:59] [INFO] Installing OpenVINO dependencies
[2026-02-28 12:07:19] [OK] Installed OpenVINO dependencies
[2026-02-28 12:07:19] [INFO] Installing dependencies
[2026-02-28 12:08:25] [OK] Dependencies Installed
[2026-02-28 12:08:25] [INFO] Installing Mise
[2026-02-28 12:08:30] [OK] Installed Mise
[2026-02-28 12:08:30] [INFO] Configuring Debian Testing Repo
[2026-02-28 12:08:33] [OK] Configured Debian Testing repo
[2026-02-28 12:08:33] [INFO] Installing packages from Debian Testing repo
[2026-02-28 12:08:34] [OK] Installed packages from Debian Testing repo
[2026-02-28 12:08:34] [INFO] Setup uv 0.10.7
[2026-02-28 12:08:36] [OK] Setup uv 0.10.7
[2026-02-28 12:08:36] [INFO] Setup PostgreSQL 16
[2026-02-28 12:08:53] [OK] Setup PostgreSQL 16
[2026-02-28 12:08:54] [INFO] Fetching GitHub release: VectorChord (0.5.3)
[2026-02-28 12:08:56] [OK] Deployed: VectorChord (0.5.3)
[2026-02-28 12:08:56] [ERROR] in line 153: exit code 2 (Misuse of shell builtins (e.g. syntax error)): while executing command sed -i -e "/^#shared_preload/s/^#//;/^shared_preload/s/''/'vchord.so'/" /etc/postgresql/16/main/postgresql.conf

  ✖️   Installation failed in container 106 (exit code: 2)
 📋 Installation log: /tmp/immich-106-dc9133ef.log

  Error: Misuse of shell builtins (e.g. syntax error)

### 🖼️ Additional context (optional).

_No response_

I tracked this down to the sed command in `immich-install.sh`

Failing line:

```
sed -i -e "/^#shared_preload/s/^#//;/^shared_preload/s/''/'vchord.so'/" /etc/postgresql/16/main/postgresql.conf
```

On my Proxmox VE 9.1.1 / Debian 13 / PostgreSQL 16 / Immich install, this consistently fails during the VectorChord step with:

```
[ERROR] in line 153: exit code 2 (Misuse of shell builtins (e.g. syntax error)): while executing command sed -i -e "/^#shared_preload/s/
  ^#//;/^shared_preload/s/''/'vchord.so'/" /etc/postgresql/16/main/postgresql.conf
```

Replacing that line with this fixed the issue locally:

```
sed -i -E "s@^#?shared_preload_libraries = .*@shared_preload_libraries = 'vchord.so'@" /etc/postgresql/16/main/postgresql.conf
```

This makes the installer write the intended PostgreSQL setting directly instead of trying to transform the existing line with the broken expression.

--- a/install/immich-install.sh
+++ b/install/immich-install.sh
@@ -150,7 +150,7 @@ PG_VERSION="16" PG_MODULES="pgvector" setup_postgresql
 VCHORD_RELEASE="0.5.3"
 fetch_and_deploy_gh_release "VectorChord" "tensorchord/VectorChord" "binary" "${VCHORD_RELEASE}" "/tmp" "postgresql-16-vchord_*_amd64.deb"
 
-sed -i -e "/^#shared_preload/s/^#//;/^shared_preload/s/''/'vchord.so'/" /etc/postgresql/16/main/postgresql.conf
+sed -i "s/^#shared_preload.*/shared_preload_libraries = 'vchord.so'/" /etc/postgresql/16/main/postgresql.conf
 systemctl restart postgresql.service
 PG_DB_NAME="immich" PG_DB_USER="immich" PG_DB_GRANT_SUPERUSER="true" PG_DB_SKIP_ALTER_ROLE="true" setup_postgresql_db
 

